main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -201,6 +214,12 @@ object BuildServerProtocol {
         })
       SourcesItem(id, items)
     },
+    bspBuildTargetResourcesItem := {
+      val id = bspTargetIdentifier.value
+      val dirs = unmanagedResourceDirectories.value
+      val managed = managedResources.value
+      ResourcesItem(id, dirs.toVector.map(_.toURI) ++ managed.toVector.map(_.toURI))
```

Comment:
Here, some resource directories are mixed with some other resource files, which does not seem right. The specification says it should contain "a list of resource file". But, if we do so we lose the relative path of each resource which is probably needed by IntelliJ when it is copied to the target directory.

So I think the specification of the `ResourceRequest` is incomplete, see #181

Reply:
You're right, the sources items have a `kind` attribute that can make a difference between a source file or directory ! 

I had a look at the intellij scala plugin : it seems that the resources can be successfully imported by [BspProjectResolver](https://github.com/JetBrains/intellij-scala/blob/idea212.x/bsp/src/org/jetbrains/bsp/project/importing/BspProjectResolver.scala#L280) . The only thing here is that I didn't know where the [BuildServerCapabilities.resourcesProvider](https://github.com/build-server-protocol/build-server-protocol/blob/master/bsp4j/src/main/xtend-gen/ch/epfl/scala/bsp4j/BuildServerCapabilities.java#L26) was set so I tried setting it manually to true in the intellij scala plugin to check that the resources were successfully imported in my project.

Reply:
Answering my own question : I need to update [bsp.contra](https://github.com/sbt/sbt/blob/develop/protocol/src/main/contraband/bsp.contra#L203) in the current PR to update the `resourcesProvider` flag